src/software-device.cpp

```diff
@@ -67,13 +68,16 @@ namespace librealsense
 
     std::shared_ptr<matcher> software_device::create_matcher(const frame_holder& frame) const
     {
-        std::vector<stream_interface*> profiles;
+        using stream_uid = std::pair< rs2_stream, int >;
+        std::set< stream_uid > streams;
+        std::vector< stream_interface * > stream_interfaces;
 
-        for (auto&& s : _software_sensors)
-            for (auto&& p : s->get_stream_profiles())
-                profiles.push_back(p.get());
+        for( auto const & s : _software_sensors )
+            for( auto const & p : s->_sw_profiles )
```

Comment:
Calling `get_raw_stream_profiles()` is clearer

Reply:
It doesn't exist any more :)

Reply:
There is `get_stream_profiles()` or something like it but it returns a vector, not a reference. Since we're a software device and these are software sensors, I elected to access directly.